### ISSUE TITLE
Horizontal sprite mosaic is inaccurate

### ISSUE THREAD

**Issue by fleroviux** [mgba-emu/mgba#2933]
I wrote a basic [test ROM](https://github.com/nba-emu/hw-test/tree/master/ppu/sprite-hmosaic) to figure out how horizontal sprite mosaic might work and currently it seems to be quite broken in mGBA.

![image](https://github.com/mgba-emu/mgba/assets/13669774/3e9b8775-ce65-450d-8591-7152c5d3c2c7)
On the left side is mGBA and on the right side is what it looks like on hardware.

The mosaic algorithm implemented in hardware seems to use an internal register to store the the sprite pixel that is actually displayed. For each pixel at x-coordinate `line_x` in the scanline the PPU looks at the sprite pixel for the current x-coordinate and decides based on three conditions if the current sprite pixel should be latched into the register. The conditions are as follows:
1. `(line_x % mosaic_width) == 0` **or**
2. the current sprite pixel has the mosaic-flag cleared **or**
3. the latched sprite pixel has the mosaic-flag cleared

In essence the first condition (likely implemented with a counter) creates the classical pixelation effect while the last two conditions handle the transitions at the boundaries of sprites, i.e. from no-sprite to sprite-with-mosaic and sprite-without-mosaic to sprite-with-mosaic and vice versa.

Finally it should be noted that the PPU updates the mosaic-flag in the sprite buffer even for transparent pixels (like with the sprite priority bug) which likely matters for this test ROM.

**Comment by fleroviux** [comment]
Yes, here's the raw output. Generated from NBA, but from what I can tell it's pixel-identical to real hardware (I should really get a proper way to capture output from real hardware some day...). 

![output](https://github.com/mgba-emu/mgba/assets/13669774/2ca3fabd-3d72-4c85-ad14-4282a305d972)

**Comment by profi200** [comment]
Here you have it dumped from a 3DS in GBA mode. Note that colors might not be accurate due to dithering and color correction. But the mosaic effect should be 1:1 accurate with GBA (SP) hardware.
![sprite-hmosaic_3DS](https://github.com/mgba-emu/mgba/assets/7831477/ca95f116-47d1-4b98-b12b-d1ad3df38bb5)

**Comment by fleroviux** [comment]
I updated the test ROM from the original post with new test cases. @Arisotura found out on the DS, that the latching behaviour also is affected by sprite priority! I was able to confirm that this also is the case on GBA. 

Basically there is an additional condition where the current sprite pixel is latched: When the priority of the current sprite pixel is *numerically* lower than the latched sprite pixel.
